cloud-controller-manager/do/resources.go

```diff
@@ -0,0 +1,142 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package do
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/digitalocean/godo"
+	"github.com/golang/glog"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	v1informers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	v1lister "k8s.io/client-go/listers/core/v1"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+)
+
+const (
+	controllerSyncPeriod     = 1 * time.Minute
+	requestTimeout           = 10 * time.Second
+	resourceTypeLoadBalancer = "load_balancer"
+)
+
+// ResourcesController ensures that DO resources are properly tagged.
+type ResourcesController struct {
+	clusterIDTag string
+	kclient      kubernetes.Interface
+	gclient      *godo.Client
+	svcLister    v1lister.ServiceLister
+}
+
+// NewResourcesController returns a new services controller responsible for
+// re-applying cluster IDs on DO load-balancers.
+func NewResourcesController(clusterIDTag string, inf v1informers.ServiceInformer,
+	k kubernetes.Interface, g *godo.Client) *ResourcesController {
+	return &ResourcesController{
+		clusterIDTag: clusterIDTag,
+		kclient:      k,
+		gclient:      g,
+		svcLister:    inf.Lister(),
+	}
+}
+
+// Run starts the resources controller. It watches over DigitalOcean resources
+// making sure the right tags are set.
+func (r *ResourcesController) Run(stopCh <-chan struct{}) {
+	ticker := time.NewTicker(controllerSyncPeriod)
+	defer ticker.Stop()
+	// Do not wait for initial tick to pass; run immediately for reduced sync
+	// latency.
+	for tickerC := ticker.C; ; {
+		if err := r.sync(); err != nil {
+			glog.Errorf("failed to sync load-balancer tags: %s", err)
+		}
+		select {
+		case <-tickerC:
+			continue
+		case <-stopCh:
+			return
+		}
+	}
+}
+
+func (r *ResourcesController) sync() error {
+	// Fetch all load-balancers.
+	ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
+	defer cancel()
+	lbs, _, err := r.gclient.LoadBalancers.List(ctx, &godo.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list load-balancers: %s", err)
+	}
+
+	// Collect tag resources for known load balancer names (i.e., services
+	// with type=LoadBalancer.)
+	svcs, err := r.svcLister.List(labels.Everything())
+	if err != err {
+		return fmt.Errorf("failed to list services: %s", err)
+	}
+	var res []godo.Resource
+	for _, lb := range lbs {
+		for _, svc := range svcs {
+			if svc.Spec.Type == corev1.ServiceTypeLoadBalancer && cloudprovider.GetLoadBalancerName(svc) == lb.Name {
+				res = append(res, godo.Resource{
+					ID:   lb.ID,
+					Type: godo.ResourceType(resourceTypeLoadBalancer),
```

Comment:
You should make a PR for godo to add Load Balancer resource type

Reply:
Yep, that's on the TODO list. 👍

Reply:
https://github.com/digitalocean/godo/pull/206